server/sonar-server/src/main/java/org/sonar/server/qualitygate/RegisterQualityGates.java

```diff
@@ -37,67 +48,205 @@
+    new QualityGateCondition().setMetricKey(NEW_RELIABILITY_RATING_KEY).setOperator(OPERATOR_GREATER_THAN).setPeriod(LEAK_PERIOD).setErrorThreshold(A_RATING),
+    new QualityGateCondition().setMetricKey(NEW_MAINTAINABILITY_RATING_KEY).setOperator(OPERATOR_GREATER_THAN).setPeriod(LEAK_PERIOD).setErrorThreshold(A_RATING),
+    new QualityGateCondition().setMetricKey(NEW_COVERAGE_KEY).setOperator(OPERATOR_LESS_THAN).setPeriod(LEAK_PERIOD).setErrorThreshold("80"),
+    new QualityGateCondition().setMetricKey(NEW_DUPLICATED_LINES_DENSITY_KEY).setOperator(OPERATOR_GREATER_THAN).setPeriod(LEAK_PERIOD).setErrorThreshold("3")
+  );
 
   private final DbClient dbClient;
   private final QualityGateConditionsUpdater qualityGateConditionsUpdater;
-  private final LoadedTemplateDao loadedTemplateDao;
   private final QualityGates qualityGates;
+  private final QualityGateDao qualityGateDao;
+  private final QualityGateConditionDao qualityGateConditionDao;
 
-  public RegisterQualityGates(DbClient dbClient, QualityGateConditionsUpdater qualityGateConditionsUpdater,
-    LoadedTemplateDao loadedTemplateDao, QualityGates qualityGates) {
+  public RegisterQualityGates(DbClient dbClient, QualityGateConditionsUpdater qualityGateConditionsUpdater, QualityGates qualityGates) {
     this.dbClient = dbClient;
     this.qualityGateConditionsUpdater = qualityGateConditionsUpdater;
-    this.loadedTemplateDao = loadedTemplateDao;
     this.qualityGates = qualityGates;
+    this.qualityGateDao = dbClient.qualityGateDao();
+    this.qualityGateConditionDao = dbClient.gateConditionDao();
   }
 
   @Override
   public void start() {
     try (DbSession dbSession = dbClient.openSession(false)) {
-      if (shouldRegisterBuiltinQualityGate(dbSession)) {
-        createBuiltinQualityGate(dbSession);
-        registerBuiltinQualityGate(dbSession);
-        dbSession.commit();
+      QualityGateDto builtin = qualityGateDao.selectByName(dbSession, BUILTIN_QUALITY_GATE_NAME);
+      if (builtin == null) {
+        LOGGER.info("Built-in quality gate [{}] has been created", BUILTIN_QUALITY_GATE_NAME);
+        builtin = createQualityGate(dbSession, BUILTIN_QUALITY_GATE_NAME);
+        if (qualityGates.getDefault() == null) {
+          qualityGates.setDefault(dbSession, builtin.getId());
+        }
       }
+
+      updateQualityConditionsIfRequired(dbSession, builtin);
+
+      qualityGateDao.ensureOneBuiltInQualityGate(dbSession, BUILTIN_QUALITY_GATE_NAME);
+
+      dbSession.commit();
     }
   }
 
-  @Override
-  public void stop() {
-    // do nothing
-  }
+  private void updateQualityConditionsIfRequired(DbSession dbSession, QualityGateDto builtin) {
+    List<QualityGateCondition> qualityGateConditions = qualityGateConditionDao.selectForQualityGate(dbSession, builtin.getId())
+      .stream()
+      .map(dto -> {
+        MetricDto metricDto = dbClient.metricDao().selectById(dbSession, dto.getMetricId());
```

Comment:
This could be optimized by loading all metrics used in conditions at once (by creating a map of metrics by ids)

Reply:
I thought about that but I was thinking that it was over engineering, but since you think the same way, I'll do.